tools/exposure-client/main.go

```
@@ -110,76 +91,24 @@ func main() {
 	if *twice {
 		time.Sleep(1 * time.Second)
 		log.Printf("sending the request again...")
-		sendRequest(jsonData)
-	}
-}
-
-func randIntervalCount() int32 {
-	n, err := rand.Int(rand.Reader, big.NewInt(144))
-	if err != nil {
-		log.Fatalf("rand.Int: %v", err)
+		sendRequest(data)
 	}
-	return int32(n.Int64() + 1) // valid values are 1-144
 }
 
 func randomInt(maxValue int) int {
-	n, err := rand.Int(rand.Reader, big.NewInt(int64(maxValue)))
-	if err != nil {
-		log.Fatalf("rand.Int: %v", err)
-	}
-	return int(n.Int64())
+	return enclient.RandomInt(maxValue)
 }
 
 func randomArrValue(arr []string) string {
 	return arr[randomInt(len(arr))]
 }
 
-func generateExposureKeys(numKeys, tr int) []database.ExposureKey {
-	keys := make([][]byte, numKeys)
-	for i := 0; i < numKeys; i++ {
-		keys[i] = make([]byte, dkLen)
-		_, err := rand.Read(keys[i])
-		if err != nil {
-			log.Fatalf("rand.Read: %v", err)
-		}
-	}
-
-	// When publishing multiple keys - they'll be on different days.
-	intervalCount := randIntervalCount()
-	intervalNumber := int32(time.Now().Unix()/600) - intervalCount
-	exposureKeys := make([]database.ExposureKey, numKeys)
-	for i, rawKey := range keys {
-		transmissionRisk := tr
-		if transmissionRisk < 0 {
-			transmissionRisk = randomInt(maxTransmissionRisk) + 1
-		}
-
-		exposureKeys[i].Key = base64.StdEncoding.EncodeToString(rawKey)
-		exposureKeys[i].IntervalNumber = intervalNumber
-		exposureKeys[i].IntervalCount = intervalCount
-		exposureKeys[i].TransmissionRisk = transmissionRisk
-		// Adjust interval math for next key.
-		intervalCount = randIntervalCount()
-		intervalNumber -= intervalCount
-	}
-	return exposureKeys
-}
-
-func sendRequest(jsonData []byte) {
-	r, err := http.NewRequest("POST", *url, bytes.NewBuffer(jsonData))
+func sendRequest(data interface{}) {
```

Same note, you probably want to bubble this error up to the top caller.